src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -901,7 +903,8 @@ mod tests {
         {
             // The frequency difference is within tolerance.
             let mut state = orig_state.clone();
-            state.tsc_khz = Some(state.tsc_khz.unwrap() + (TSC_KHZ_TOL / 2.0).round() as u32);
+            state.tsc_khz =
+                Some(state.tsc_khz.unwrap() + TSC_KHZ_TOL_NUMERATOR / TSC_KHZ_TOL_DENOMINATOR / 2);
```

Comment:
While rewriting this code, I noticed that `TSC_KHZ_TOL` is small enough that `TSC_KHZ_TOL_NUMERATOR / TSC_KHZ_TOL_DENOMINATOR / 2` will round to 0; I don't think that's intended behavior for the test code. This looks like it's been part of the existing code for some time; `(TSC_KHZ_TOL / 2.0).round()` would also round to 0. The same applies to two pieces of code below.

Reply:
... yeah, you're right, this makes very little sense to me, too.  The `* 2` case is actually even more baffling, because there's a comment that says "The frequency difference is over the tolerance.", but then the assertion still asserts that we're not above the tolerance (which we are indeed not, because we add 0 to our value). I'm assuming that these tests meant to do something like

```rs
state = Some(state.tsc_khz.unwrap() + state.tsc_khz.unwrap() * (TSC_KHZ_TOL / 2))
```

and that the second test is meant to have its assertion inverted. Nice catch!

cc @kalyazin

Reply:
that is indeed a good finding. What Patrick proposes makes sense to me.

Reply:
When I was refactoring this code I also encountered this, I thought it looked weird at the time.

Reply:
I don't really think we need a new test, we can just fix this one in this PR